pkg/cloud/openstack/clients/machineservice.go

```diff
@@ -71,12 +76,71 @@ type InstanceListOpts struct {
 	Name string `q:"name"`
 }
 
+func GetCloudFromSecret(kubeClient kubernetes.Interface, namespace string, secretName string, cloudName string) (clientconfig.Cloud, error) {
+	emptyCloud := clientconfig.Cloud{}
+
+	if cloudName == "" {
+		cloudName = "openstack"
```

Comment:
The issue is that if you don't have `openstack` as the name for your cloud and you are having issues because you didn't set the `cloudName`, you won't easily be able to trace or discover that issue. 
If you are completely against erroring here then there needs to be some logging that clearly states `cloudName` was unset and that we are defaulting it to `openstack`

Reply:
done